client/dto/Analytics.cs

```diff
@@ -4,43 +4,83 @@
+    // 1. Evaluation metrics.
+    // 2. Target metrics.
+    // Both types inherit from this base class.
+    public abstract class Analytics : IEquatable<Analytics>
     {
-        private readonly FeatureConfig featureConfig;
-        private readonly Target target;
-        private readonly Variation variation;
+        protected readonly Target target;
 
-        public FeatureConfig FeatureConfig { get => featureConfig; }
-        public Target Target { get => target; }
-        public Variation Variation { get => variation; }
-
-        public Analytics(FeatureConfig featureConfig, Target target, Variation variation, EventType eventType)
+        protected Analytics(Target target)
         {
-            this.featureConfig = featureConfig;
             this.target = target;
-            this.variation = variation;
         }
 
+        public Target Target => target;
+        public abstract bool Equals(Analytics other);
+
         public override bool Equals(object obj)
         {
             return Equals(obj as Analytics);
         }
 
-        public bool Equals(Analytics other)
+        public abstract override int GetHashCode();
+    }
+
+// EvaluationAnalytics subclass
+    public class EvaluationAnalytics : Analytics
+    {
+        // The global target is used when we don't want to use the actual target in the evaluation metrics
+        // payload. 
+        public static readonly string GlobalTargetIdentifier = "__global__cf_target";
+        public static readonly string GlobalTargetName = "Global Target";
+
+        public EvaluationAnalytics(FeatureConfig featureConfig, Variation variation, Target target)
+            : base(target)
+        {
+            FeatureConfig = featureConfig;
+            Variation = variation;
+        }
+
+        public FeatureConfig FeatureConfig { get; }
+
+        public Variation Variation { get; }
+
+
+        public override bool Equals(Analytics other)
         {
-            return other != null &&
-                   EqualityComparer<FeatureConfig>.Default.Equals(featureConfig, other.featureConfig) &&
-                   EqualityComparer<Target>.Default.Equals(target, other.target) &&
-                   EqualityComparer<Variation>.Default.Equals(variation, other.variation);
+            return other is EvaluationAnalytics otherEvaluation
+                   && EqualityComparer<FeatureConfig>.Default.Equals(FeatureConfig, otherEvaluation.FeatureConfig)
+                   && EqualityComparer<Variation>.Default.Equals(Variation, otherEvaluation.Variation);
         }
 
         public override int GetHashCode()
         {
-            int hashCode = -1526478203;
-            hashCode = hashCode * -1521134295 + EqualityComparer<FeatureConfig>.Default.GetHashCode(featureConfig);
-            hashCode = hashCode * -1521134295 + EqualityComparer<Target>.Default.GetHashCode(target);
-            hashCode = hashCode * -1521134295 + EqualityComparer<Variation>.Default.GetHashCode(variation);
```

Comment:
Hash code comparison isn't using target. Let me fix. This wasn't caught in testing because the global target is always sent, so it's the same effect as not comparing it here.